### ISSUE TITLE
`global-whitespace-mode` is enabled but `whitespace-mode` is disabled

### ISSUE THREAD

**Issue by mangkoran** [hlissner/doom-emacs#5856]
### What did you expect to happen?

`whitespace-mode` is enabled globally using `global-whitespace-mode`

### What actually happened?

`whitespace-mode` is `nil` even though `global-whitespace-mode` is `t`

### Describe your attempts to resolve the issue

Based on https://github.com/hlissner/doom-emacs/issues/2673, I have the following lines in `config.el`:
```elisp
(use-package! whitespace
  :config
  (setq
    whitespace-style '(face spaces tabs newline space-mark tab-mark newline-mark )
    whitespace-display-mappings '(
      (space-mark   ?\     [?\u00B7]     [?.])
      (space-mark   ?\xA0  [?\u00A4]     [?_])
      (newline-mark ?\n    [?¬ ?\n])
      (tab-mark     ?\t    [?\u00BB ?\t] [?\\ ?\t]))))

(global-whitespace-mode +1)
```
But every time I open any file in Doom, `whitespace-mode` is `nil`.

I also provide `whitespace-enable-predicate` var here (I did not set it tho), just in case it might be useful:
```
#[128 "\300\301\"\205\0\300\302\"\207"
      [apply doom-is-childframes-p
             #[0 "\303=\204$\0<\203 \0\242\304=\203\0	A>?\202!\0	>\202!\0\305\205A\0\n?\205A\0\306 \307H\310=?\205A\0\306 \307H\311=?\206A\0\306 \312\230\207"
                 [whitespace-global-modes major-mode noninteractive t not nil buffer-name 0 32 42 "*scratch*"]
                 2]
             nil]
      4 nil]
```
### Steps to reproduce

1. Put the above config lines in `config.el`
2. Reopen Doom
3. Open any file
4. `whitespace-mode` should be `nil` and `global-whitespace-mode` should be `t`

### System Information

https://pastebin.com/0fsnDTfG

### Notes

Might be related to https://github.com/hlissner/doom-emacs/issues/5854, but the OP did not provide enough information.

**Comment by hlissner** [comment]
As it turns out, `global-whitespace-mode` and `whitespace-mode` operate independently. The global mode bypasses the local mode and activates its underlying machinery directly, so testing `whitespace-mode` for a non-nil value won't actually tell you if it is in effect or not.

That said, I also discovered that my childframe predicate function was incorrectly negated, which prevents `global-whitespace-mode` from actually kicking in anywhere _but_ child frames. I've fixed that in eefa0ec, which should resolve this issue (and #5854, for that matter).

Let me know if that isn't the case and I'll reopen this.
